### ISSUE TITLE
Remove the examples category "Our Favorite Recipies"

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#18300]
https://matplotlib.org/devdocs/gallery/index.html#our-favorite-recipes

"Favorite" is very subjective. And to me this is just a random selection, nothing particularly intriguing or noteworthy.

The examples from this category should either be removed (if redunant) or moved to other sections.

**Comment by diegoleal** [comment]
I studied and looked around the [gallery](https://matplotlib.org/devdocs/gallery/index.html) for a bit, here's what I propose to be done with [out favorite recipes](https://matplotlib.org/devdocs/gallery/index.html#our-favorite-recipes):

- The [Placing text boxes](https://matplotlib.org/devdocs/gallery/recipes/placing_text_boxes.html) is almost identical to the [Pyplot text](https://matplotlib.org/devdocs/gallery/pyplots/pyplot_text.html), the only difference is that the former uses a `bbox` and the second don't. I suggest merging the two together (even a deletion can be considered, there are a lot of box examples already).

- The [Common Date Problems](https://matplotlib.org/devdocs/gallery/recipes/common_date_problems.html) is almost identical to [Formatting date ticks using ConciseDateFormatter](https://matplotlib.org/devdocs/gallery/ticks_and_spines/date_concise_formatter.html#sphx-glr-gallery-ticks-and-spines-date-concise-formatter-py). The latter is a little more verbose but it introduces the `ConciseDateFormatter`. I believe `Common Date Problems` can be deleted.

- [Centered Spines With Arrows](https://matplotlib.org/devdocs/gallery/recipes/centered_spines_with_arrows.html) is almost identical to [Spine Placement Demo](https://matplotlib.org/devdocs/gallery/ticks_and_spines/spine_placement_demo.html). The only difference is that the first one implements arrow tips. For the sake of making the examples more distinguishable (some users might want to learn just how to make arrow tips) I suggest keeping both and moving `Centered Spine With Arrows` to the `Ticks and Spines` section.

- [Sharing Axis Limits and Views](https://matplotlib.org/devdocs/gallery/recipes/share_axis_lims_views.html) is identical to [Shared Axis Demo](https://matplotlib.org/devdocs/gallery/subplots_axes_and_figures/shared_axis_demo.html) and the latter is more detailed. I suggest deleting `Sharing Axis Limits and Views`.

- [Fill Between Alpha](https://matplotlib.org/devdocs/gallery/recipes/fill_between_alpha.html) is a tough call for me. The example is big (3 examples in one html page) but it's well explained and it covers some interesting math phenomena like random walks which makes for a good read overall, OTOH, there are at least 6 other fill examples that do more or less the same. I suggest moving to either `Miscellaneous` or `Lines, bars and markers` section, but a deletion could be considered.

I could start the implementations, but, if possible, I would like to know what someone more experienced think of this suggestions before starting.

**Comment by story645** [comment]
* placing text labels is using OO while pyplot is using `pyplot` so I don't know about merging the two interfaces, is there an OO example if can be merged into? (xref:  #15922 'cause part of the issue is that the pyplot section is inconsistent)
* slightly conflicted b/c this is more general than conciseDateFormatter, but ConciseDateFormatter probably fixes the problem for most people (see #17283)
* agree with our suggestion for moving that example
* this is another pyplot/OO problem, agree on deletion but than either sharedAxisDemo should be updated to be more OO or a pure OO example should be spun up to replace Sharing Axis Limits and Views (& possibly use https://matplotlib.org/3.3.1/users/whats_new.html#new-axes-sharex-axes-sharey-methods)
* agree on moving to lines, bars, and markers with the rest of the fill_between demos and leave consolidation of those demos to someone else.
